desktop/libs/notebook/src/notebook/connectors/sql_alchemy_tests.py

```diff
@@ -243,6 +243,28 @@ def test_create_engine_with_impersonation(self):
       create_engine.assert_called_with('presto://test@hue:8080/hue', pool_pre_ping=True)
 
 
+  def test_explain(self):
+
+    with patch('notebook.connectors.sql_alchemy.SqlAlchemyApi._create_connection') as _create_connection:
+      with patch('notebook.connectors.sql_alchemy.SqlAlchemyApi._create_engine') as _create_engine:
+        with patch('notebook.connectors.sql_alchemy.SqlAlchemyApi._get_session') as _get_session:
+
+          result = [{"id": 1, "select_type": "SIMPLE"}, {"Extra": "No tables used"}]
+          
+          execute = Mock(return_value=result)
+          _create_connection.return_value = Mock(
+            execute=execute
+          )
+          notebook = {}
+          snippet = {'statement': 'SELECT 1;'}
+
+          explanation = "\n".join("{}: {},".format(k, v) for row in result for k, v in row.items())
```

Comment:
Shouldn't it be a raw string instead of redoing the same code logic?

Reply:
done